src/Compilers/Core/Portable/Compilation/IExpression.cs

```diff
@@ -795,31 +795,21 @@ public interface IObjectCreationExpression : IExpression
         /// <summary>
         /// Explicitly-specified member initializers.
         /// </summary>
-        ImmutableArray<IMemberInitializer> MemberInitializers { get; }
+        ImmutableArray<ISymbolInitializer> MemberInitializers { get; }
     }
 
     /// <summary>
     /// Represents an object member initializer.
     /// </summary>
-    public interface IMemberInitializer : IOperation
+    public interface ISymbolInitializer : IOperation
     {
-        MemberInitializerKind MemberInitializerKind { get; }
         IExpression Value { get; }
     }
-
-    /// <summary>
-    /// Kinds of member initializers.
-    /// </summary>
-    public enum MemberInitializerKind
-    {
-        Field,
-        Property
-    }
-
+    
     /// <summary>
     /// Represents an initialization of a field.
     /// </summary>
-    public interface IFieldInitializer : IMemberInitializer
+    public interface IFieldInitializer : ISymbolInitializer
```

Comment:
OK, I'm about to push a new commit that replaces the InitializedField property with an InitializedFields property. This makes it possible to encode all of the fields declared with an As New declaration. However, it doesn't work end to end because MemberSemanticModel and InitializerSemanticModel conspire to bake in an assumption that an initializer applies to precisely one symbol. I'm not going to fix this as part of this change set, but I can open an issue for it.

Also, my assumption is that there is no way for a property initializer to apply to more than one property, despite at least some of the VB compiler's implementation being prepared for that. @AlekseyTs, is this assumption correct?